engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -193,19 +197,44 @@ public Rope getLiteralSource() {
    * @param source the module source.
    */
   public void setLiteralSource(String source) {
-    setLiteralSource(Rope.apply(source));
+    setLiteralSource(Rope.apply(source), null, null);
   }
 
   /**
    * Sets new literal sources for the module.
    *
    * @param source the module source.
+   * @param change suggested small change in a single literal
+   * @param edit description of the small edit made
    */
-  public void setLiteralSource(Rope source) {
-    this.literalSource = source;
+  public void setLiteralSource(Rope source, IR.Literal change, model.TextEdit edit) {
```

Comment:
How thread-safe this method is supposed to be? Are we assuming it is always going to be access only by a single thread, @kustosz ?

Reply:
Yes it is safe to assume single thread access. We ensure that there are no parallel compilations happening